services/src/datasets/external/netcdfcf/database.rs

```diff
@@ -0,0 +1,1200 @@
+                group_names,
+                entity_id,
+                meta_data
+            ) VALUES (
+                $1,
+                $2,
+                $3 :: TEXT[],
+                $4,
+                $5 :: "GdalMetaDataList"
+            );"#,
+            &[
+                tokio_postgres::types::Type::UUID,
+                tokio_postgres::types::Type::TEXT,
+                tokio_postgres::types::Type::TEXT_ARRAY,
+                tokio_postgres::types::Type::INT8,
+                // we omit `GdalMetaDataList`
+            ],
+        )
+        .await
+        .boxed_context(error::CannotStoreLoadingInfo)?;
+
+    for loading_info in loading_infos {
+        transaction
+            .execute(
+                &loading_info_stmt,
+                &[
+                    &provider_id,
+                    &file_name,
+                    &loading_info.group_path,
+                    &(loading_info.entity_id as i64),
+                    &loading_info.meta_data,
+                ],
+            )
+            .await
+            .boxed_context(error::CannotStoreLoadingInfo)?;
+    }
+
+    Ok(())
+}
+
+pub(crate) async fn remove_overviews(
+    transaction: &Transaction<'_>,
+    provider_id: DataProviderId,
+    file_name: &str,
+) -> Result<bool> {
+    let removed_rows = transaction
+        .execute(
+            "
+            DELETE FROM
+                ebv_provider_overviews
+            WHERE
+                provider_id = $1 AND
+                file_name = $2
+            ",
+            &[&provider_id, &file_name],
+        )
+        .await
+        .boxed_context(error::UnexpectedExecution)?;
+
+    Ok(removed_rows > 0)
+}
+
+pub struct NetCdfDatabaseListingConfig<'a> {
+    pub provider_id: DataProviderId,
+    pub collection: &'a LayerCollectionId,
+    pub file_name: &'a str,
+    pub group_path: &'a [String],
+    pub options: &'a LayerCollectionListOptions,
+}
+
+/// A flag that indicates on-going process of an overview.
+///
+/// Cleans up the flag if dropped.
+pub struct InProgressFlag<D: NetCdfCfProviderDb> {
+    db: Arc<D>,
+    provider_id: DataProviderId,
+    file_name: String,
+}
+
+impl<D: NetCdfCfProviderDb> InProgressFlag<D> {
+    pub async fn create(
+        db: Arc<D>,
+        provider_id: DataProviderId,
+        file_name: String,
+    ) -> Result<Self> {
+        let success = db.lock_overview(provider_id, &file_name).await?;
+
+        if !success {
+            return Err(NetCdfCf4DProviderError::CannotCreateInProgressFlag);
+        }
+
+        Ok(Self {
+            db,
+            provider_id,
+            file_name,
+        })
+    }
+
+    pub async fn remove(self) -> Result<()> {
+        self.db
+            .unlock_overview(self.provider_id, &self.file_name)
+            .await?;
+        Ok(())
+    }
+}
+
+impl<D: NetCdfCfProviderDb> Drop for InProgressFlag<D> {
+    fn drop(&mut self) {
+        let result = tokio::task::block_in_place(move || {
```

Comment:
if I run the task I get 

> thread 'actix-server worker 1' panicked at /home/michael/git/geoengine/services/src/datasets/external/netcdfcf/database.rs:1143:22:
can call blocking only when running on the multi-threaded runtime

Reply:
[885eaaf](https://github.com/geo-engine/geoengine/pull/941/commits/885eaaf58812b68bce9d79a45cf3032eb8ba3389)